.devops/deploy-pipelines.yml

```diff
@@ -329,3 +329,72 @@ stages:
                     VALUE_FILE: "helm/pay-wallet-values-uat.yaml"
                     GREEN_VERSION: $(app_version)
   # --- END Deploy UAT --- #
+
+  # --- START of Validate event in Hub (UAT) --- #
+
+  - stage: "UAT_Testing_Generate_Wallet_Event"
+    displayName: "UAT Testing Generate Wallet Event"
+    dependsOn: [ Deploy_UAT_Green]
+    pool:
+      vmImage: 'ubuntu-latest'
+    jobs:
+      - job: ValidateCDCFlow
+        displayName: 'Run API Tests DataLake Flow (UAT)'
+        steps:
+          - script: |
+              yarn global add newman
+            displayName: 'newman installation'
+
+          - script: |
+              newman run --ignore-redirects api-tests/wallet_payment_api.cards.tests.json --environment=api-tests/uat.cards.envs.json --env-var WALLET_TOKEN_TEST=$(WALLET_TOKEN_TEST) --reporters cli,junit,json > output.txt
+            displayName: 'Run API tests (UAT)'
+
+          - script: |
+              # Extract WALLET_ID from report JSON
+              WALLET_ID=$(grep -oP "WALLET_ID=\K[^\']+" output.txt)
+              
+              if [ -z "$WALLET_ID" ]; then
+                echo "Cannot extract WALLET_ID from report"
+                exit 1
+              fi
+              echo "WALLET_ID=$WALLET_ID"
+              echo "##vso[task.setvariable variable=WALLET_ID]$WALLET_ID"
+            displayName: 'Extract and export WALLET_ID from JSON report'
+
+          - script: |
+              wget https://archive.apache.org/dist/kafka/2.8.0/kafka_2.13-2.8.0.tgz
+              tar -xzf kafka_2.13-2.8.0.tgz
+              export PATH=$PATH:$PWD/kafka_2.13-2.8.0/bin
+              # Verify Kafka installation
+              kafka-topics.sh --version
+              sleep 4
+
+              # Run the Kafka consumer command
+              eventsFromKafka=$(kafka-console-consumer.sh \
+                --bootstrap-server pagopa-u-itn-observ-evh.servicebus.windows.net:9093 \
+                --topic payment-wallet-ingestion-dl \
+                --from-beginning \
```

Comment:
I considered both this scenarios, but initially i decided to accept the risk as i considered a race condition in a UAT  post-deployment scenario pretty rare. But yes, it could happen.

I understand your idea but maybe even with parallel process running the race condition could still happen.
The only way to never have it is to (as you suggested) filter alle events against the WALLET_ID 

i will add this feature right now,
is that ok?